src/tk/wurst_client/mods/BaseFinderMod.java

```diff
@@ -60,94 +69,51 @@ public void onUpdate()
 		if(hasTimePassedM(3000))
 		{
 			matchingBlocks.clear();
+			
+			int playerX = (int) Math.floor(mc.thePlayer.posX);
+			int playerY = (int) Math.floor(mc.thePlayer.posY);
+			int playerZ = (int) Math.floor(mc.thePlayer.posZ);
+
 			for(int y = range; y >= -range; y--)
-			{
 				for(int x = range; x >= -range; x--)
-				{
 					for(int z = range; z >= -range; z--)
 					{
-						int posX = (int)(mc.thePlayer.posX + x);
-						int posY = (int)(mc.thePlayer.posY + y);
-						int posZ = (int)(mc.thePlayer.posZ + z);
+						int posX = playerX + x;
```

Comment:
btw: JAVA IS NOT SLOW.
It can almost be as fast as C/C++ and sometime even faster!
Modenrn jvm's have very good just in time compiler.
it will do optimization based on runtime profiling.
e.g. Escaped analysis allows to detect pure data structures classes.
The jvm won't create instances of those and pass the values directly to the corresponding place.